pinot-core/src/main/java/org/apache/pinot/core/operator/timeseries/TimeSeriesAggregationOperator.java

```diff
@@ -240,14 +242,12 @@ public void processStringExpression(BlockValSet blockValSet, Map<Long, BaseTimeS
     }
   }
 
-  public static long[] applyTimeshift(long timeshift, long[] timeValues, int numDocs) {
-    if (timeshift == 0) {
-      return timeValues;
+  private void applyTimeOffset(long[] timeValues, int numDocs) {
```

Comment:
Method only has name change correct ?

Reply:
We also do the time-shift in-place instead of allocating another array.